region/lowernorfair/east/Fast Pillars Setup Room.json

```diff
@@ -1605,45 +1634,74 @@
       "flashSuitChecked": true
     },
     {
-      "id": 52,
+      "id": 64,
       "link": [4, 4],
-      "name": "Carry G-Mode Back Through",
+      "name": "Base",
       "entranceCondition": {
-        "comeInWithGMode": {
-          "mode": "direct",
-          "morphed": false
-        }
+        "comeInNormally": {}
       },
-      "requires": [],
-      "exitCondition": {
-        "leaveWithGMode": {
-          "morphed": false
-        }
+      "requires": [
+        {"or": [
+          "canCarefulJump",
+          {"enemyDamage": {
+            "enemy": "Yellow Space Pirate (wall)",
+            "type": "contact",
+            "hits": 1
+          }}
+        ]},
+        {"heatFrames": 275}
+      ],
+      "flashSuitChecked": true,
+      "note": "Fall onto the platform, then wait for the pirate to jump back to the right before returning through the door."
+    },
+    {
+      "id": 65,
+      "link": [4, 4],
+      "name": "Land in Door Frame",
+      "entranceCondition": {
+        "comeInNormally": {}
       },
-      "bypassesDoorShell": true,
-      "flashSuitChecked": true
+      "requires": [
+        "canPrepareForNextRoom",
+        "canInsaneJump",
+        {"heatFrames": 80},
+        {"doorUnlockedAtNode": 4}
+      ],
+      "unlocksDoors": [
+        {"types": ["super"], "requires": []},
+        {"types": ["missiles", "powerbomb"], "requires": ["never"]}
+      ]
     },
     {
-      "id": 53,
       "link": [4, 4],
-      "name": "Carry G-Mode Morph Back Through",
+      "name": "Door Stuck, Open Door",
       "entranceCondition": {
-        "comeInWithGMode": {
-          "mode": "direct",
-          "morphed": true
-        }
+        "comeInWithDoorStuckSetup": {}
       },
       "requires": [
-        "canArtificialMorph"
+        "canPrepareForNextRoom",
+        {"heatFrames": 80},
```

Comment:
Before, the doorstuck and come in jumping strats were combined and shared the 80 frames before. I'm not sure if this is actually accurate though.

Reply:
I had 54 frame room times for both doorstuck and comeInJumping.